src/libs/kata-types/src/config/runtime.rs

```diff
@@ -0,0 +1,276 @@
+    pub enable_pprof: bool,
+
+    /// Determines whether container seccomp profiles are passed to the virtual machine and
+    /// applied by the kata agent. If set to true, seccomp is not applied within the guest.
+    #[serde(default)]
+    pub disable_guest_seccomp: bool,
+
+    /// Determines how VFIO devices should be be presented to the container.
+    ///
+    /// Options:
+    /// - vfio: Matches behaviour of OCI runtimes (e.g. runc) as much as possible.  VFIO devices
+    ///   will appear in the container as VFIO character devices under /dev/vfio. The exact names
+    ///   may differ from the host (they need to match the VM's IOMMU group numbers rather than
+    ///   the host's)
+    /// - guest-kernel: This is a Kata-specific behaviour that's useful in certain cases.
+    ///   The VFIO device is managed by whatever driver in the VM kernel claims it. This means
+    ///   it will appear as one or more device nodes or network interfaces depending on the nature
+    ///   of the device. Using this mode requires specially built workloads that know how to locate
+    ///   the relevant device interfaces within the VM.
+    #[serde(default)]
+    pub vfio_mode: String,
+
+    /// Vendor customized runtime configuration.
+    #[serde(default, flatten)]
+    pub vendor: RuntimeVendor,
+}
+
+impl ConfigOps for Runtime {
+    fn adjust_configuration(conf: &mut TomlConfig) -> Result<()> {
+        RuntimeVendor::adjust_configuration(conf)?;
+        if conf.runtime.internetworking_model.is_empty() {
+            conf.runtime.internetworking_model = default::DEFAULT_INTERNETWORKING_MODEL.to_owned();
+        }
+
+        for bind in conf.runtime.sandbox_bind_mounts.iter_mut() {
+            resolve_path!(*bind, "sandbox bind mount `{}` is invalid: {}")?;
+        }
+
+        Ok(())
+    }
+
+    fn validate(conf: &TomlConfig) -> Result<()> {
+        RuntimeVendor::validate(conf)?;
+
+        let net_model = &conf.runtime.internetworking_model;
+        if !net_model.is_empty()
+            && net_model != "macvtap"
+            && net_model != "none"
+            && net_model != "tcfilter"
+        {
+            return Err(eother!(
+                "Invalid internetworking_model `{}` in configuration file",
+                net_model
+            ));
+        }
+
+        let vfio_mode = &conf.runtime.vfio_mode;
+        if !vfio_mode.is_empty() && vfio_mode != "vfio" && vfio_mode != "guest-kernel" {
+            return Err(eother!(
+                "Invalid vfio_mode `{}` in configuration file",
+                vfio_mode
+            ));
+        }
+
+        for bind in conf.runtime.sandbox_bind_mounts.iter() {
+            validate_path!(*bind, "sandbox bind mount `{}` is invalid: {}")?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Runtime {
+    /// Check whether experiment `feature` is enabled or not.
+    pub fn is_experiment_enabled(&self, feature: &str) -> bool {
+        for f in self.experimental.iter() {
```

Comment:
Can be simplified by `self.experimental.contains(&feature.to_string())`

Reply:
fixed in next version